onnxruntime/hosting/server_configuration.h

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef ONNXRUNTIME_HOSTING_SERVER_CONFIGURATION_H
+#define ONNXRUNTIME_HOSTING_SERVER_CONFIGURATION_H
+
+#include <thread>
+#include <fstream>
+
+#include "boost/program_options.hpp"
+
+namespace onnxruntime {
+namespace hosting {
+
+namespace po = boost::program_options;
+
+// Wrapper around Boost program_options and should provide all the functionality for options parsing
+// Provides sane default values
+class ServerConfiguration {
+ public:
+  ServerConfiguration() {
+    desc.add_options()("help,h", "Shows a help message and exits");
+    desc.add_options()("model_path,m", po::value(&model_path)->required(), "Path to ONNX model");
+    desc.add_options()("address,a", po::value(&address)->default_value(address), "The base HTTP address");
+    desc.add_options()("port,p", po::value(&port)->default_value(port), "HTTP port to listen to requests");
+    desc.add_options()("threads,t", po::value(&threads)->default_value(threads), "Number of http threads");
+  }
+
+  void ParseInput(int ac, char** av) {
```

Comment:
One of the way to verify the negative case is to make this function return `true`/`false` other than exit the program. Then the caller need to decide whether exit or not.

Reply:
But how would they know to `exit(EXIT_SUCCESS` or `exit(EXIT_FAILURE)`?

Reply:
true means success, false means failed :-)

Reply:
There's three scenarios:
1. EXIT_SUCCESS
2. EXIT_FAILURE
3. We want to continue

In one of these commits I returned an int and ran `exit` in main but EXIT_SUCCESS and EXIT_FAILURE can be different values so wanting to continue was difficult to convey and made for an ugly API.

Reply:
Now, we could just use a bool and only ever use `exit(EXIT_SUCESS)` in the main but we lose the ability to use EXIT_FAILURE when there's a parsing failure.